node/process.ts

```diff
@@ -216,11 +216,10 @@ class Process extends EventEmitter {
   constructor() {
     super();
 
-    //This causes the exit event to be binded to the unload event
+    // This causes the exit event to be binded to the unload event
     globalThis.addEventListener("unload", () => {
-      //TODO(Soremwar)
-      //Get the exit code from the unload event
-      super.emit("exit", 0);
+      super.emit("exit", process.exitCode);
+      Deno.exit(process.exitCode || 0);
```

Comment:
This is probably wrong - if there are more "unload" listeners attached they will never be run. Opened the PR to get discussion going.

Reply:
@bnoordhuis any ideas how we tackle that except adding functionality equivalent to `process.exitCode` directly into CLI?

Reply:
> This is probably wrong - if there are more "unload" listeners attached they will never be run. Opened the PR to get discussion going.

I have pushed a fix for this to the PR.